design/resources.go

```diff
@@ -122,6 +122,28 @@ var _ = a.Resource("workitem.2", func() {
 		a.Response(d.InternalServerError)
 	})
 
+	a.Action("update", func() {
+		a.Security("jwt")
+		a.Routing(
+			a.PATCH("/:id"),
+		)
+		a.Description("update the given work item with given id.")
+		a.Params(func() {
+			a.Param("id", d.String, "id")
+		})
+		a.Payload(updateWorkItemJSONAPIPayload)
+		a.Response(d.OK, func() {
+			// Still using workitem in MediaTypes.
+			// ToDo update to struct which complies to jsonapi
+			a.Media(workItem2)
+		})
+		a.Response(d.BadRequest, func() {
```

Comment:
We later need to change the responses like so, once my PR (#421) gets merged:

```go
		a.Response(d.BadRequest, JSONAPIErrors)
		a.Response(d.InternalServerError, JSONAPIErrors)
		a.Response(d.NotFound, JSONAPIErrors)
		a.Response(d.Unauthorized, JSONAPIErrors)
```

This is because all JSONAPI endpoints must/should return this. This is just a reminder.